colossus-testkit/src/main/scala/colossus-testkit/FakeIOSystem.scala

```diff
@@ -29,19 +30,52 @@ object FakeIOSystem {
     (sys, probe)
   }
 
-  /**
-   * Returns an entirely in-thread CallbackExecutor
-   */
-  def callingThreadCallbackExecutor(implicit system: ActorSystem): CallbackExecutor = {
-    class FakeExecutor extends Actor with CallbackExecution {
-      def receive = handleCallback
+  //prevents accidentally spinning up a new executor per test
+  private val exCache: collection.mutable.Map[ActorSystem, CallbackExecutor] = collection.mutable.Map()
+
+  def testExecutor(implicit system: ActorSystem): CallbackExecutor = {
+    exCache.get(system).getOrElse {
+      val ref = system.actorOf(Props[GenericExecutor].withDispatcher("server-dispatcher"))
```

Comment:
Is there any case to be made for making a generic-executor-dispatcher, which is pinned?  I know server-dispatcher will probably never change, but not sure if we want to make it pull double duty.  I'm fine with keeping it, but at least wanted to bring up the point